iree-prof-tools/iree-vis.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <string>
+
+#include "compiler/bindings/c/iree/compiler/embedding_api.h"
+#include "compiler/bindings/c/iree/compiler/mlir_interop.h"
+#include "third_party/abseil-cpp/absl/base/log_severity.h"
+#include "third_party/abseil-cpp/absl/log/globals.h"
+#include "third_party/abseil-cpp/absl/log/initialize.h"
+#include "third_party/abseil-cpp/absl/log/log.h"
+#include "third_party/abseil-cpp/absl/flags/flag.h"
+#include "third_party/abseil-cpp/absl/flags/parse.h"
+#include "third_party/abseil-cpp/absl/strings/str_cat.h"
+#include "third_party/llvm-project/mlir/include/mlir/IR/Operation.h"
+
+ABSL_FLAG(std::string, input_iree_file, "",
+          "IREE MLIR text assembly file to read.");
+ABSL_FLAG(std::string, iree_phase, "executable-sources",
+          "IREE compilation phase of input IREE MLIR file.");
+
+void Walk(mlir::Operation* op, absl::string_view indent) {
+  LOG(INFO) << indent << op->getName().getStringRef().str();
+  std::string child_indent = absl::StrCat(indent, "  ");
+  op->walk([op, indent = absl::string_view(child_indent)](
+      mlir::Operation* nested_op) {
+    if (nested_op != op) {
+      Walk(nested_op, indent);
+    }
+  });
+}
+
+int main(int argc, char** argv) {
+  absl::SetStderrThreshold(absl::LogSeverityAtLeast::kInfo);
+  absl::InitializeLog();
+  absl::ParseCommandLine(argc, argv);
+
+  ireeCompilerGlobalInitialize();
+  auto* session = ireeCompilerSessionCreate();
+
+  std::string input_iree_file = absl::GetFlag(FLAGS_input_iree_file);
```

Comment:
Let's not include the dumping logic in the tool.

In that way, we can make the tool simple and do one simple thing well, which is to convert an MLIR into graph json (or some other visualization format).

Reply:
Dumping is just to check whether parsing is correct. It will be replaced with json output logic in a folloing PR.

Reply:
Let's do this later. I understand your pain in the build system.